src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/PermissionsManager.java

```diff
@@ -166,20 +168,43 @@ public List<String> getPermissions(Clan clan) {
         return permissions.get(clan.getTag());
     }
 
+
     /**
      * Charge a player some money
-     *
+     * @deprecated Use playerChargeMoney(OfflinePlayer, double, Cause) instead
      */
+    @Deprecated(since = "2.19.0", forRemoval = true)
     public boolean playerChargeMoney(OfflinePlayer player, double money) {
-        return economy.withdrawPlayer(player, money).transactionSuccess();
+        return playerChargeMoney(player, money, null);
     }
 
     /**
-     * Grants a player some money
+     * Charge a player some money
      *
      */
+    public boolean playerChargeMoney(OfflinePlayer player, double money, @Nullable Cause cause) {
+        EconomyTransactionEvent event = new EconomyTransactionEvent(player, money, cause, EconomyTransactionEvent.TransactionType.WITHDRAW_FROM_PLAYER);
+        Bukkit.getPluginManager().callEvent(event);
+        return economy.withdrawPlayer(player, event.getAmount()).transactionSuccess();
+    }
+
+    /**
+     * Grants a player some money
+     * @deprecated Use playerGrantMoney(OfflinePlayer, double, Cause) instead
+     */
+    @Deprecated(since = "2.19.0", forRemoval = true)
     public boolean playerGrantMoney(OfflinePlayer player, double money) {
-        return economy.depositPlayer(player, money).transactionSuccess();
+        return playerGrantMoney(player, money, null);
+    }
+
+    /**
+     * Grants a player some money
+     *
+     */
+    public boolean playerGrantMoney(OfflinePlayer player, double money, @Nullable Cause cause) {
+        EconomyTransactionEvent event = new EconomyTransactionEvent(player, money, cause, EconomyTransactionEvent.TransactionType.DEPOSIT_TO_PLAYER);
+        Bukkit.getPluginManager().callEvent(event);
+        return economy.depositPlayer(player, event.getAmount()).transactionSuccess();
```

Comment:
One more thing, if a transaction was failed or cancelled by some reason, this method returns `false`.
And we're using it to `return` in [some cases](https://github.com/RoinujNosde/SimpleClans/pull/384/files#diff-31ec9c53bbd78dc4516615be348ce16b6c5ae7f215691562ec26b52cb7ff3b29R229). it means, that your event would be called even if transaction is invalid. Also I think that some developers may want to cancel the EconomyTransactionEvent event.

Reply:
I have maded the event cancellable. I do the event after the transaction, to check if is invalid or not. If the amount has changed, or if the event results as cancelled, a new transaction will be done, but now the thing is... what about if the new transaction fails? I ignore the result, because if the event is cancelled, the action has to be cancelled.